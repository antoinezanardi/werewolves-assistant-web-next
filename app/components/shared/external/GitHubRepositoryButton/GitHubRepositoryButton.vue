<template>
  <a
    id="github-repository-link"
    class="p-button primary"
    href="https://www.github.com/antoinezanardi/werewolves-assistant-web-next"
    rel="noopener noreferrer"
    target="_blank"
  >
    <FontAwesomeIcon
      icon="fab fa-github"
    />

    <span>
      {{ textButton }}
    </span>
  </a>
</template>

<script lang="ts" setup>
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";

import type { GitHubRepositoryButtonProps } from "~/components/shared/external/GitHubRepositoryButton/git-hub-repository-button.types";

const props = defineProps<GitHubRepositoryButtonProps>();

const { t } = useI18n();

const textButton = computed<string>(() => {
  if (props.textButton !== undefined) {
    return props.textButton;
  }
  return t("components.GitHubRepositoryButton.thisProjectIsOpenSource");
});
</script>