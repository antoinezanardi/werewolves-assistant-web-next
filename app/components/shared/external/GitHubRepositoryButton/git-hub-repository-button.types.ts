type GitHubRepositoryButtonProps = {
  textButton?: string;
};

export type { GitHubRepositoryButtonProps };