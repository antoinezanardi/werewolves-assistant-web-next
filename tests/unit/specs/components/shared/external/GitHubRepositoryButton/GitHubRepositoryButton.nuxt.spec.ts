import type { mount } from "@vue/test-utils";

import type { GitHubRepositoryButtonProps } from "~/components/shared/external/GitHubRepositoryButton/git-hub-repository-button.types";
import GitHubRepositoryButton from "~/components/shared/external/GitHubRepositoryButton/GitHubRepositoryButton.vue";
import { mountSuspendedComponent } from "~/tests/unit/utils/helpers/mount.helpers";

describe("GitHub Repository Button", () => {
  let wrapper: ReturnType<typeof mount<typeof GitHubRepositoryButton>>;
  const defaultProps: GitHubRepositoryButtonProps = { textButton: "GitHub" };

  beforeEach(async() => {
    wrapper = await mountSuspendedComponent(GitHubRepositoryButton, { props: defaultProps });
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Button Text", () => {
    it("should render button text with other text when text is passed as prop.", async() => {
      wrapper = await mountSuspendedComponent(GitHubRepositoryButton, { props: { textButton: "Other text" } });
      const buttonText = wrapper.find<HTMLAnchorElement>("#github-repository-link");

      expect(buttonText.text()).toBe("Other text");
    });

    it("should render button text with default text when text is not passed as prop.", async() => {
      wrapper = await mountSuspendedComponent(GitHubRepositoryButton);
      const buttonText = wrapper.find<HTMLAnchorElement>("#github-repository-link");

      expect(buttonText.text()).toBe("components.GitHubRepositoryButton.thisProjectIsOpenSource");
    });
  });
});