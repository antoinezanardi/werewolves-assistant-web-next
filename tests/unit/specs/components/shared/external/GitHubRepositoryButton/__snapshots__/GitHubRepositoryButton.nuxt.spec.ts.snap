// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GitHub Repository Button > should render component and match snapshot when mounted. 1`] = `"<a id="github-repository-link" class="p-button primary" href="https://www.github.com/antoinezanardi/werewolves-assistant-web-next" rel="noopener noreferrer" target="_blank"><span class="fa-github fab me-2"></span><span>GitHub</span></a>"`;
